src/helpers.js

```diff
@@ -873,3 +873,41 @@
 export function canDeleteDiskFile(disk) {
     return disk.type === "volume" || (disk.type === "file" && disk.source.file);
 }
+
+export function getStoragePoolPath(storagePools, poolName, connectionName) {
+    const pool = storagePools.find(pool => pool.name === poolName && pool.connectionName === connectionName);
+
+    return pool?.target?.path;
+}
+
+export function vmSupportsExternalSnapshots(config, vm, storagePools) {
+    const disks = Object.values(vm.disks);
+
+    // If at leat one disk has internal snapshot preference specified, use internal snapshot for all disk,
+    // as mixing internal and external is not allowed
+    if (disks.some(disk => disk.snapshot === "internal")) {
+        logDebug(`vmSupportsExternalSnapshots: vm ${vm.name} has internal snapshot preference specified, false`);
+        return false;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240129-095224-b2f949bc-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)